app/components/PostFormModal.tsx

```diff
@@ -0,0 +1,90 @@
+'use client'
+import { useEffect, useRef, useState, FormEvent } from 'react';
+import styles from "../../styles/Modal.module.css";
+
+export default function PostFormModal() {
+  const [showModal, setShowModal] = useState<Boolean>(false)
+  const modalRef = useRef<null | HTMLDialogElement>(null);
+
+  useEffect(() => {
+    if (showModal) {
+      modalRef.current?.showModal();
+    } else {
+      modalRef.current?.close();
+    }
+  }, [showModal]);
+
+  /**
+   *  Opens the modal
+   */
+  function clickOpen(): void {
+    setShowModal(true);
+  }
+
+  /**
+   * Closes the modal
+   */
+  function clickClose(): void {
+    setShowModal(false);
+  }
+
+  /**
+   * Submits new post data then closes the modal
+   * 
+   * @param e - the form submit event
+  */
+  function clickOk(e: FormEvent<HTMLFormElement>): void {
```

Comment:
Rename to something more intuitive. `clickOk` doesn't accurately describe what the function is used for. Maybe `onSubmitHandler` or something similar?